### PR TITLE
III-4601 Made format error message RFC7807 compliant for duplicate places

### PR DESCRIPTION
**Pull request by grubolsch**
### Changed
I Made format error message RFC7807 compliant for duplicate places
I now throw ApiProblem for this new format.

Minor bonus: changed all the status codes in ApiProblem to use constants.

Related: https://github.com/cultuurnet/apidocs/pull/359

---
Ticket: https://jira.uitdatabank.be/browse/III-4601

### CODE REVIEW

File: features/place/duplicate.feature [hunk 1]
```diff
@@ -11,9 +11,15 @@ Feature: Test creating places
     Given I create a random name of 6 characters and keep it as "name"
     Given I create a minimal place and save the "id" as "originalPlaceId" then I should get a "201" response code
     Then I wait for the place with url "/places/%{originalPlaceId}" to be indexed
-    Given I create a minimal place and save the "originalPlace" as "newPlaceUri" then I should get a "409" response code
-    Then the JSON response at "originalPlace" should be:
+    Given I create a minimal place and save the "query" as "query" then I should get a "409" response code
```
Comment (LucWollants) [inline features/place/duplicate.feature:14]: Is it still needed to save this `query` value? I get the feeling that we could do without this extra custom step by just calling the existing steps like, for example:

```
When I send a POST request to "/places/"
Then the response status should be "409"
And I keep the value of the JSON response at "id" as "originalPlaceId"
```

File: src/Http/Place/ImportPlaceRequestHandler.php [hunk 2]
```diff
@@ -283,4 +263,26 @@ public function handle(ServerRequestInterface $request): ResponseInterface
...
+        if (! $this->preventDuplicatePlaces) {
+            return;
+        }
+
+        try {
+            $duplicatePlaceId = $this->lookupDuplicatePlace->getDuplicatePlaceUri($place);
+            if ($duplicatePlaceId !== null) {
+                throw ApiProblem::statusConflict(
+                    MultipleDuplicatePlacesFound::ERROR_MSG,
+                    ['query' => $duplicatePlaceId]
```
Comment (LucWollants) [inline src/Http/Place/ImportPlaceRequestHandler.php:278]: When there is only one duplicate place, is it then still a query?

File: tests/Http/Place/ImportPlaceRequestHandlerTest.php [hunk 3]
```diff
@@ -5197,55 +5205,66 @@ public function testHandleWithDuplicatePreventionEnabled(Repository $aggregateRe
             ->withJsonBodyFromArray($this->getSimplePlace())
             ->build('POST');
 
+        $this->expectExceptionObject(ApiProblem::statusConflict(
+            MultipleDuplicatePlacesFound::ERROR_MSG,
+            ['instance' => $originalPlaceUri]
```
Comment (LucWollants) [inline tests/Http/Place/ImportPlaceRequestHandlerTest.php:5210]: Here `instance` is still used but the guard throws `query`, which makes me think there is still something else going on.

File: src/Http/Place/ImportPlaceRequestHandler.php [hunk 4]
```diff
@@ -283,4 +263,26 @@ public function handle(ServerRequestInterface $request): ResponseInterface
...
+    {
+        if (! $this->preventDuplicatePlaces) {
+            return;
+        }
+
+        try {
+            $duplicatePlaceId = $this->lookupDuplicatePlace->getDuplicatePlaceUri($place);
+            if ($duplicatePlaceId !== null) {
+                throw ApiProblem::statusConflict(
+                    MultipleDuplicatePlacesFound::ERROR_MSG,
```
Comment (LucWollants) [inline src/Http/Place/ImportPlaceRequestHandler.php:277]: It feels weird to get the error message from the `MultipleDuplicatePlacesFound` although there is exactly one place returned as duplicate.

File: src/Http/Place/ImportPlaceRequestHandler.php (hunk 2 above)
Comment (grubolsch) [reply src/Http/Place/ImportPlaceRequestHandler.php:279]: What would you prefer as the param name?

Comment (LucWollants) [reply src/Http/Place/ImportPlaceRequestHandler.php:279]: Something like the parameter and method name, maybe`duplicatePlaceUri`?

File: tests/Http/Place/ImportPlaceRequestHandlerTest.php (hunk 3 above)
Comment (LucWollants) [reply tests/Http/Place/ImportPlaceRequestHandlerTest.php]: This is not verified with `expectExceptionObject` I think it is better to use `AssertApiProblemTrait`

Comment (grubolsch) [reply tests/Http/Place/ImportPlaceRequestHandlerTest.php]: done

File: src/Http/Place/ImportPlaceRequestHandler.php (hunk 2 above)
Comment (grubolsch) [reply src/Http/Place/ImportPlaceRequestHandler.php]: changed it

**Review by LucWollants — APPROVED** [review]

**Pull request merged**